moveit_kinematics/ikfast_kinematics_plugin/scripts/auto_create_ikfast_moveit_plugin.sh

```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+
+# Create an ikfast plugin package for MoveIt from a given robot URDF
+# This includes:
+# - conversion of URDF to Collada
+# - running openrave from a docker image to create the actual ikfast solver
+# - running create_ikfast_moveit_plugin.py to create the corresponding plugin package
+
+function print_help {
+	echo "$(basename $0) [--help|-h] [--quiet|-q] [--keep] [--name|-n] [--pkg|-p] [--iktype|-t] <input> <group> <base> <eef>"
+	echo "  input          .urdf, .dae, or .cpp input file"
+	echo "                 File type determines required processing stages."
+	echo "  group          planning group"
+	echo "  base           name of base link"
+	echo "  eef            name of end-effector link"
+	echo "--quiet          suppress output of OpenRave"
+	echo "--keep           keep intermediate results, i.e. don't delete temporary folder on exit"
+	echo "--name <robot>   Robot name, default extracted from urdf"
+	echo "--pkg  <name>    Package name, default: <robot>_<planning group>_ikfast_plugin"
+	echo "--iktype <type>  OpenRave kinematics type [default: Transform6D]"
+}
+
+function parse_options {
+	# set defaults (if not set in environment yet)
+	IK_TYPE=${IK_TYPE:-Transform6D}
+	QUIET=${QUIET:-0}
+	KEEP_RESULTS=${KEEP_RESULTS:-0}
+	while true ; do
+		case "$1" in
+			--help|-h)
+				print_help
+				exit 0
+				;;
+			--quiet|-q)
+				QUIET=1
+				;;
+			--keep)
+				KEEP_RESULTS=1
+				;;
+			--iktype|-t)
+				IK_TYPE=$2; shift
+				;;
+			--name|-n)
+				ROBOT_NAME=$2; shift
+				;;
+			--pkg|-p)
+				PKG_NAME=$2; shift
+				;;
+			-*)
+				echo "Unknown option: $1"
+				print_help
+				exit 1
+				;;
+			*) break ;;
+		esac
+		shift
+	done
+	if [ $# -ne 4 ] ; then
+		echo "Expecting 4 positional arguments! Got $#: $*"
+		print_help
+		exit 1
+	fi
+	INPUT=$1
+	PLANNING_GROUP=$2
+	BASE_LINK=$3
+	EEF_LINK=$4
+}
+
+function set_option_defaults {
+	# ROBOT_NAME is auto-extracted from URDF or otherwise needs to be specified as an option
+	if [ -z "$ROBOT_NAME" ] ; then
+		echo "Undefined robot name. Please specify option --name".
+		exit 1
+	fi
+	# Define default PKG_NAME if not yet defined
+	PKG_NAME=${PKG_NAME:-$ROBOT_NAME_$PLANNING_GROUP_ikfast_plugin}
+}
+
+function cleanup {
+	rm -rf "$TMP_DIR"
+}
+
+function build_docker_image {
+	test "$__DOCKER_BUILT" == "1" && return
+
+	echo "Building docker image"
+	cat <<EOF > $TMP_DIR/Dockerfile
+FROM personalrobotics/ros-openrave
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends python-pip && \
+    apt-get install -y --no-install-recommends build-essential liblapack-dev ros-indigo-collada-urdf || true && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+# enforce a specific version of sympy, which is known to work with OpenRave
+RUN pip install sympy==0.7.1
```

Comment:
Not sure, what this option actually does. But we need essentially a downgrade!
See Gijs' answer [here](https://answers.ros.org/question/263925/generating-an-ikfast-solution-for-4-dof-arm).

Reply:
Note that OpenRAVE is quite the moving target, and in my experience tutorials / howtos for using it or one of its components are almost out-of-date the minute they are published. So it could well be that the `sympy` downgrade is no longer needed.

I know I needed it at the time I wrote that answer though.

Reply:
Thanks for the hint. Will check.

Reply:
$ `pip2 install --user sympy==0.7.1`
```
Collecting sympy==0.7.1
Installing collected packages: sympy
Successfully installed sympy-1.3
```

Reply:
I remember there being an issue with this in the past but I just verified that this command works as expected. I assume that there must have been other issues with my python path setup back then. No need to change this

Reply:
I had previously found that pip will skip a file if it already has a newer version. That said, I tested this locally and the command worked fine as is.

Reply:
@gavanderhoorn, the downgrade is still needed on the docker image.

Reply:
Might want to add a note to the comment there. Something along the lines of "check this again in the future, may not be needed any more at some point".